### ISSUE TITLE
Dottydoc needs to be reactive

### ISSUE THREAD

**Issue by felixmulder** [lampepfl/dotty#1945]
The doc tool currently pretty much only supports desktop, since there are no rules for mobile devices.

Since the documentation is using bootstrap 4, we should adapt to using its preferred method of handling reactive sizing.

**Comment by hhandoko** [comment]
I might be able to help 😄

**Comment by felixmulder** [comment]
Let me know if you need help getting started.

**Comment by hhandoko** [comment]
Yup, actually I do 😁  How do I (re)generate the existing Dottydocs to preview the changes?

I've tried to running `dottydoc` on the root project but it's throwing up some `[doc error]` messages (not being able to find the HTML pages). Is there a specific sub-project / module I need to switch to?

So here's what I've accomplished so far:

1. I've cloned the repo, compiled and managed to get to the REPL successfully
1. I've located the relevant files under `doc-tool/resources`

**Comment by felixmulder** [comment]
Run `genDocs` and you'll get a `_site` directory in the docs dir :)

**Comment by julienrf** [comment]
@felixmulder I think you mean “responsive” rather than “reactive” ;)

**Comment by felixmulder** [comment]
Yes random ahoy buzzword, I mean responsive

**Comment by hhandoko** [comment]
@felixmulder I've submitted the first PR for the responsive feature [1]. I'll be looking into the landing page next. Are there specific areas of the site / layout you want me to look at in particular?

Notes:
[1] - I prefer a few smaller PRs rather than a single large one as it is easier to manage. Hope that's alright.

**Comment by felixmulder** [comment]
So the important part for getting dottydoc responsive is not the landing page - that page is going to be different for every project. The import part is getting the pages under: http://dotty.epfl.ch/docs/ to be responsive :)

**Comment by hhandoko** [comment]
Cool, got it 👌

**Comment by hhandoko** [comment]
@felixmulder any particular area you want me to look next? Otherwise I'm going to update the docs (`api-page.html`) template styling, e.g. clean up extra padding, improve contrast, font sizes, etc.
